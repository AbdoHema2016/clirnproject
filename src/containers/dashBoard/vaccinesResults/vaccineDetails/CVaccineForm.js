import React, {Component} from 'react';
import {View, Platform} from 'react-native';
import {connect} from 'react-redux';
import Style from './style';
import {LOCAL_PATH, Constants, docTypes} from '../../../../utilities';
import {translate} from '../../../../Localization';

import CTitleAndInputField from '../../../../components/cTitleAndInputField';
import CButtonWithImage from '../../../../components/cButtonWithImage';
import CButton from '../../../../components/cButton';
import {ScrollView, Switch} from 'react-native-gesture-handler';
import CLabel from '../../../../components/cLabel';
import VaccineMethods from './methods';
import helperMethods from './helperFunctionsForAddVaccine';
import {
  uploadVaccineImageAction,
  uploadVaccineDetailsAction,
  showVaccineModalAction,
  setVaccineNameAction,
  setVaccineCentreNameAction,
  setVaccineAddDateAction,
  setVaccineExpireDateAction,
  setVaccineDocumentAction,
  editVaccineDetailsAction,
  resetDetailsAction,
} from '../redux/actions';
import {showVaccinePopUpOnHomeAction} from '../../Home/redux/actions';
import {connectActionSheet} from '@expo/react-native-action-sheet';
import CDatePicker from '../../../../components/cDatePicker';
import CPicker from '../../../../components/cPicker';
import CTextField from '../../../../components/cTextField';
const {testIds, fieldTypes, vaccineDateType, otherVaccineTypeKey} = Constants;
const unfinishedChanges = true;
class CVaccineForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDatePicker: false,
      dateType: '',
      checkMissingFields: false,
      modalOpen: false,
      test_date: {value: '', isEmpty: true},
      expiration_date: {value: '', isEmpty: true},
      showPicker: false,
      otherVaccineTypeChosen: false,
      otherVaccineName: '',
      otherVaccineNameError: false,
    };
    VaccineMethods.setProps(this.props);
    VaccineMethods.setInstance(this);
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevProps.vaccinations !== this.props.vaccinations) {
      VaccineMethods.setProps(this.props);
    }
  }

  uploadVaccineImage = () => {
    const options = [
      translate('CAMERA_OPTIONS.TAKE_PHOTO'),
      translate('CAMERA_OPTIONS.LIBRARY'),
      translate('CAMERA_OPTIONS.CANCEL'),
    ];
    const cancelButtonIndex = 2;
    this.props.showActionSheetWithOptions(
      {
        useModal: true,
        options,
        cancelButtonIndex,
      },
      (buttonIndex) => {
        VaccineMethods.uploadImage(buttonIndex, docTypes.VACCINE);
      },
    );
  };
  showTypePicker = (name) => {
    const {vaccineTypes} = this.props;
    if (!name.value) {
      VaccineMethods.setVaccineName(vaccineTypes[0]);
    }

    this.setState({otherVaccineTypeChosen: name.value === otherVaccineTypeKey});

    this.setState({showPicker: !this.state.showPicker});
  };
  render() {
    const {modal} = this.props;
    const {
      showDatePicker,
      checkMissingFields,
      otherVaccineTypeChosen,
      otherVaccineNameError,
    } = this.state;
    const {
      vaccineDetailsToBeEdited: {
        name,
        health_center,
        test_date,
        expiration_date,
        document,
      },
      vaccineTypes,
    } = this.props;

    return (
      <View style={Style.inputContainer}>
        <ScrollView>
          <View style={Style.inputInnerContainer}>
            {modal && (
              <CButtonWithImage
                buttonContainerStyle={Style.crossButtonContainer}
                buttonCustomStyle={Style.crsossButton}
                textStyle={Style.textStyle}
                customeImageStyle={Style.customeImageStyle}
                imagePath={LOCAL_PATH.CROSS_ICON}
                onPress={() => VaccineMethods.hideVaccineModal()}
              />
            )}
            <CTitleAndInputField
              testID={testIds.vaccineName}
              accessibilityLabel={testIds.vaccineName}
              type={Platform.OS === 'ios' ? 'Button' : 'Picker'}
              imagePath={LOCAL_PATH.DROP_DOWN_ICON}
              onPress={() => this.showTypePicker(name)}
              selectedValue={name.value}
              androidPickerContainer={modal && Style.buttonContainerStyle}
              setTitle={(title) => {
                VaccineMethods.setVaccineName(title);
                this.setState({
                  otherVaccineTypeChosen: title === otherVaccineTypeKey,
                });
              }}
              pickerItems={vaccineTypes}
              textStyle={
                !name.value ? Style.titleNillColor : Style.titleSelectedColor
              }
              buttonContainerStyle={modal && Style.buttonContainerStyle}
              buttonCustomStyle={modal && Style.buttonContainerStyle}
              text={
                name.value || translate('VACCINE_SCREEN.CHOOSE_VACCINE_TYPE')
              }
              titleStyle={Style.inputLabel}
              title={translate('VACCINE_SCREEN.VACCINE_NAME')}
              customSuperContainer={modal && Style.buttonContainerStyle}
              placeHolderText={translate('STRINGS.ENTER_NAME')}
              value={name.value}
              onChangeText={(val) => VaccineMethods.setVaccineName(val)}
              error={checkMissingFields && name.isEmpty}
              errorMessage={translate('VACCINE_SCREEN.EMPTY_VACCINE_NAME')}
            />
            {otherVaccineTypeChosen && (
              <CTextField
                customSuperContainer={modal && Style.buttonContainerStyle}
                placeHolderText={translate(
                  'VACCINE_SCREEN.PLEASE_ENTER_VACCINE_NAME',
                )}
                error={otherVaccineNameError}
                onChangeText={(val) => VaccineMethods.setOtherVaccineName(val)}
                errorMessage={translate(
                  'VACCINE_SCREEN.VACCINE_NAME_CANNOT_BE_EMPTY',
                )}
              />
            )}
            <CTitleAndInputField
              testID={testIds.vaccineCentre}
              accessibilityLabel={testIds.vaccineCentre}
              type={fieldTypes.TEXT_FIELD}
              titleStyle={Style.inputLabel}
              title={translate('VACCINE_SCREEN.CENTRE')}
              customSuperContainer={modal && Style.buttonContainerStyle}
              imagePath={LOCAL_PATH.DROP_DOWN_ICON}
              textStyle={Style.titleSelectedColor}
              placeHolderText={translate('VACCINE_SCREEN.WHERE')}
              androidPickerContainer={Style.buttonContainerStyleOnDataFilled}
              value={health_center.value}
              onChangeText={(val) => VaccineMethods.setCentreName(val)}
              error={checkMissingFields && health_center.isEmpty}
              errorMessage={translate(
                'VACCINE_SCREEN.EMPTY_HEALTH_CENTRE_NAME',
              )}
            />
            <CTitleAndInputField
              testID={testIds.vaccineDate}
              accessibilityLabel={testIds.vaccineDate}
              type={fieldTypes.BUTTON}
              titleStyle={Style.inputLabel}
              title={translate('VACCINE_SCREEN.DATE')}
              buttonContainerStyle={modal && Style.buttonContainerStyle}
              buttonCustomStyle={modal && Style.buttonContainerStyle}
              imagePath={LOCAL_PATH.DROP_DOWN_ICON}
              textStyle={Style.titleSelectedColor}
              text={
                !test_date.value
                  ? translate('STRINGS.DATE')
                  : VaccineMethods.formatDate(test_date.value)
              }
              onPress={() => {
                this.setState(
                  {
                    showDatePicker: !this.state.showDatePicker,
                    dateType: vaccineDateType.TEST,
                  },

                  () => {
                    VaccineMethods.setDate();
                  },
                );
              }}
              error={checkMissingFields && test_date.isEmpty}
              errorMessage={translate('VACCINE_SCREEN.TEST_DATE_ERROR')}
            />
            <CTitleAndInputField
              testID={testIds.vaccineExpirationDate}
              accessibilityLabel={testIds.vaccineExpirationDate}
              type={fieldTypes.BUTTON}
              titleStyle={Style.inputLabel}
              title={translate('VACCINE_SCREEN.EXPIRATION_DATE')}
              buttonContainerStyle={modal && Style.buttonContainerStyle}
              buttonCustomStyle={modal && Style.buttonContainerStyle}
              imagePath={LOCAL_PATH.DROP_DOWN_ICON}
              textStyle={Style.titleSelectedColor}
              text={
                !expiration_date.value
                  ? translate('STRINGS.DATE')
                  : VaccineMethods.formatDate(expiration_date.value)
              }
              onPress={() => {
                this.setState(
                  {
                    showDatePicker: !this.state.showDatePicker,
                    dateType: vaccineDateType.EXPIRY,
                  },

                  () => {
                    VaccineMethods.setDate();
                  },
                );
              }}
            />
            {!unfinishedChanges && (
              <View style={Style.reminderSwitchView}>
                <View style={Style.notificationLabelView}>
                  <CLabel
                    text={translate('STRINGS.NOTIFICATION')}
                    style={Style.notificationLabel}
                  />
                  <CLabel
                    text={translate('VACCINE_SCREEN.REMINDER')}
                    style={Style.reminderLabel}
                  />
                </View>
                <Switch style={Style.switchStyle} />
              </View>
            )}

            <CTitleAndInputField
              testID={testIds.imagePicker}
              accessibilityLabel={testIds.imagePicker}
              type={fieldTypes.BUTTON}
              titleStyle={Style.inputLabel}
              title={translate('VACCINE_SCREEN.DOCUMENT')}
              imagePath={LOCAL_PATH.UPLOAD_ICON}
              textStyle={Style.uploadButtonTitle}
              buttonContainerStyle={[
                Style.uploadButtonContainer,
                modal && Style.uploadButtonContainerWidthOnDataFilled,
              ]}
              customeImageStyle={Style.uploadButtonIcon}
              buttonCustomStyle={Style.uploadButton}
              text={
                !document.value
                  ? translate('VACCINE_SCREEN.UPLOAD')
                  : document.value
              }
              onPress={this.uploadVaccineImage}
              error={checkMissingFields && !document.value}
              errorMessage={translate('STRINGS.NO_IMAGE_ERROR')}
            />
            <CButton
              testID={testIds.continue}
              accessibilityLabel={testIds.continue}
              text={translate('STRINGS.SAVE')}
              textStyle={Style.whiteText}
              buttonContainerStyle={[
                Style.greenButton,
                modal && Style.greenButtonInModal,
              ]}
              onPress={VaccineMethods.save}
            />
          </View>
          {!modal && (
            <View style={Style.backButtonView}>
              <CButton
                testID={testIds.continue}
                accessibilityLabel={testIds.continue}
                text={translate('BACK.BACK_BUTTON_TITLE')}
                textStyle={Style.blackText}
                buttonContainerStyle={Style.whiteButton}
                onPress={helperMethods.goBack}
              />
            </View>
          )}
          {showDatePicker && (
            <CDatePicker
              maximumDate={
                this.state.dateType === vaccineDateType.EXPIRY
                  ? null
                  : new Date()
              }
              mode={'date'}
              showDatePicker={() => {
                this.setState({
                  showDatePicker: !this.state.showDatePicker,
                });
              }}
              setChosenDate={VaccineMethods.getDate}
              modalVisible={showDatePicker}
              chosenDate={VaccineMethods.returnChosenDate()}
            />
          )}
        </ScrollView>
        <CLabel text={this.state.showPicker} style={Style.reminderLabel} />
        {this.state.showPicker && (
          <CPicker
            setTitle={VaccineMethods.setVaccineName}
            showDatePicker={() => this.showTypePicker(name)}
            pickerItems={vaccineTypes}
            selectedValue={name.value}
          />
        )}
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  const {
    userProfile: {
      showVaccinePopUp,
      userInfo: {lastAddedVaccine},
      loadingImage,
    },
    vaccinations: {
      vaccineDetailsToBeEdited,
      uploadingImageLoader,
      showVaccineModal,
      vaccineTypes,
    },
    vaccinations,
  } = state;

  return {
    details: state.signIn,
    profile: state.userProfile,
    feeling: state.feeling,
    healthTests: state.healthTest,
    loadingImage,
    lastAddedVaccine,
    showVaccinePopUp,
    vaccineDetailsToBeEdited,
    uploadingImageLoader,
    vaccinations,
    showVaccineModal,
    vaccineTypes,
  };
};

const mapDispatchToProps = {
  uploadVaccineImage: uploadVaccineImageAction,
  uploadVaccineDetails: uploadVaccineDetailsAction,
  showVaccinePopUpOnHome: showVaccinePopUpOnHomeAction,
  showVaccineModalOnVaccineList: showVaccineModalAction,
  setVaccineName: setVaccineNameAction,
  setVaccineCentreName: setVaccineCentreNameAction,
  setVaccineAddDate: setVaccineAddDateAction,
  setVaccineExpireDate: setVaccineExpireDateAction,
  setVaccineDocument: setVaccineDocumentAction,
  editVaccineDetails: editVaccineDetailsAction,
  resetDetails: resetDetailsAction,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(connectActionSheet(CVaccineForm));
