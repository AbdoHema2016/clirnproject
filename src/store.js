import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
