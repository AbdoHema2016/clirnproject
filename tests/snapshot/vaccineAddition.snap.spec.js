import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import CVaccineForm from '../../src/containers/dashBoard/vaccinesResults/vaccineDetails/CVaccineForm';

describe('Vaccine Form', () => {
  it('initial vaccine form data', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CVaccineForm />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
