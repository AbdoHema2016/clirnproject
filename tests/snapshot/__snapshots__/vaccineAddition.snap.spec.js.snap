// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vaccine Form initial vaccine form data 1`] = `
<View
  style={
    Object {
      "height": "auto",
      "justifyContent": "center",
      "marginTop": 16,
      "paddingBottom": 16,
      "width": 750,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "marginLeft": 13,
            "width": 724,
          }
        }
      >
        <View
          style={
            Object {
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2B342C",
                  "fontFamily": "SofiaProRegular",
                  "fontSize": 12,
                  "height": 16,
                  "marginTop": 16,
                },
              ]
            }
          >
            [missing "en.VACCINE_SCREEN.VACCINE_NAME" translation]
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "height": 48,
                  "marginTop": 4,
                },
                undefined,
                null,
              ]
            }
          >
            <View />
            <Image
              source={
                Object {
                  "ReactComponent": [Function],
                }
              }
              style={
                Object {
                  "height": 9,
                  "position": "absolute",
                  "right": 18,
                  "width": 18,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2B342C",
                  "fontFamily": "SofiaProRegular",
                  "fontSize": 12,
                  "height": 16,
                  "marginTop": 16,
                },
              ]
            }
          >
            [missing "en.VACCINE_SCREEN.CENTRE" translation]
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": "auto",
                  "width": 724,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "height": 48,
                    "justifyContent": "center",
                    "marginTop": 4,
                    "width": "100%",
                  },
                  undefined,
                  null,
                ]
              }
            >
              <TextInput
                accessibilityLabel="vaccine centre"
                accessible={true}
                allowFontScaling={true}
                blurOnSubmit={true}
                numberOfLines={1}
                onChangeText={[Function]}
                placeholder="[missing \\"en.VACCINE_SCREEN.WHERE\\" translation]"
                placeholderTextColor="#8D8C91"
                rejectResponderTermination={true}
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontSize": 18,
                      "fontWeight": "500",
                      "lineHeight": 20,
                      "paddingLeft": 19,
                      "paddingRight": 19,
                      "width": "100%",
                    },
                    Object {},
                  ]
                }
                testID="vaccine centre"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2B342C",
                  "fontFamily": "SofiaProRegular",
                  "fontSize": 12,
                  "height": 16,
                  "marginTop": 16,
                },
              ]
            }
          >
            [missing "en.VACCINE_SCREEN.DATE" translation]
          </Text>
          <View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "height": 48,
                  },
                  null,
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 4,
                      "height": 52,
                      "justifyContent": "center",
                      "shadowColor": "#DDDDDD",
                      "shadowOffset": Object {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.75,
                      "shadowRadius": 4,
                    },
                    Object {
                      "height": "100%",
                      "shadowColor": "rgba(0,0,0,0)",
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="vaccine date"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderRadius": 4,
                      "height": "100%",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 724,
                    }
                  }
                  testID="vaccine date"
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontFamily": "SofiaProRegular",
                          "fontSize": 17,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#2B342C",
                          "left": 16,
                          "position": "absolute",
                        },
                      ]
                    }
                  >
                    [missing "en.STRINGS.DATE" translation]
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "height": 18,
                            "position": "absolute",
                            "right": 18,
                            "width": 18,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <FastImageView
                      resizeMode="contain"
                      source={
                        Object {
                          "cache": "immutable",
                          "uri": undefined,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2B342C",
                  "fontFamily": "SofiaProRegular",
                  "fontSize": 12,
                  "height": 16,
                  "marginTop": 16,
                },
              ]
            }
          >
            [missing "en.VACCINE_SCREEN.EXPIRATION_DATE" translation]
          </Text>
          <View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "height": 48,
                  },
                  null,
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 4,
                      "height": 52,
                      "justifyContent": "center",
                      "shadowColor": "#DDDDDD",
                      "shadowOffset": Object {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.75,
                      "shadowRadius": 4,
                    },
                    Object {
                      "height": "100%",
                      "shadowColor": "rgba(0,0,0,0)",
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="vaccine expiration date"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderRadius": 4,
                      "height": "100%",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 724,
                    }
                  }
                  testID="vaccine expiration date"
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontFamily": "SofiaProRegular",
                          "fontSize": 17,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#2B342C",
                          "left": 16,
                          "position": "absolute",
                        },
                      ]
                    }
                  >
                    [missing "en.STRINGS.DATE" translation]
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "height": 18,
                            "position": "absolute",
                            "right": 18,
                            "width": 18,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <FastImageView
                      resizeMode="contain"
                      source={
                        Object {
                          "cache": "immutable",
                          "uri": undefined,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": "auto",
              "width": "auto",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2B342C",
                  "fontFamily": "SofiaProRegular",
                  "fontSize": 12,
                  "height": 16,
                  "marginTop": 16,
                },
              ]
            }
          >
            [missing "en.VACCINE_SCREEN.DOCUMENT" translation]
          </Text>
          <View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "height": 48,
                  },
                  null,
                  Array [
                    Object {
                      "borderStyle": "dotted",
                      "borderWidth": 1,
                      "justifyContent": "center",
                      "marginTop": 8,
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 4,
                      "height": 52,
                      "justifyContent": "center",
                      "shadowColor": "#DDDDDD",
                      "shadowOffset": Object {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.75,
                      "shadowRadius": 4,
                    },
                    Object {
                      "height": "100%",
                      "shadowColor": "rgba(0,0,0,0)",
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="ImagePicker"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": undefined,
                      "borderRadius": 4,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": "auto",
                    }
                  }
                  testID="ImagePicker"
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontFamily": "SofiaProRegular",
                          "fontSize": 17,
                          "textAlign": "center",
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#2B342C",
                          "fontSize": 18,
                          "maxWidth": "75%",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    [missing "en.VACCINE_SCREEN.UPLOAD" translation]
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "height": 18,
                            "position": "absolute",
                            "right": 18,
                            "width": 18,
                          },
                          Object {
                            "marginLeft": 32,
                            "position": "relative",
                          },
                        ],
                      ]
                    }
                  >
                    <FastImageView
                      resizeMode="contain"
                      source={
                        Object {
                          "cache": "immutable",
                          "uri": undefined,
                        }
                      }
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "shadowColor": "#DDDDDD",
                "shadowOffset": Object {
                  "height": 8,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 4,
              },
              Array [
                Object {
                  "backgroundColor": "#02BF8E",
                  "marginTop": 35,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            accessibilityLabel="Continue"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderRadius": 4,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": 718,
              }
            }
            testID="Continue"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "SofiaProRegular",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                ]
              }
            >
              [missing "en.STRINGS.SAVE" translation]
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "shadowColor": "#DDDDDD",
                "shadowOffset": Object {
                  "height": 8,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 4,
              },
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFFFFF",
                "elevation": 2,
                "marginTop": 16,
                "shadowColor": "#8D8C91",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 724,
              },
            ]
          }
        >
          <View
            accessibilityLabel="Continue"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderRadius": 4,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": 718,
              }
            }
            testID="Continue"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "SofiaProRegular",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#2B342C",
                  },
                ]
              }
            >
              [missing "en.BACK.BACK_BUTTON_TITLE" translation]
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Text
    style={
      Array [
        Object {
          "color": "#2B342C",
          "fontFamily": "arial",
          "fontSize": 16,
          "marginTop": 8,
        },
      ]
    }
  />
</View>
`;
